src/phylum/ci/ci_gitlab.py

```diff
@@ -35,17 +42,38 @@ def _check_prerequisites(self) -> None:
         """
         super()._check_prerequisites()
 
+        # References:
+        # https://github.com/watson/ci-info/blob/master/vendors.json
+        # https://docs.gitlab.com/ee/ci/variables/predefined_variables.html
         if os.getenv("GITLAB_CI") != "true":
             raise SystemExit(" [!] Must be working within the GitLab CI environment")
 
         # A GitLab token with API access is required to use the API (e.g., to post notes/comments).
         # This can be a personal, project, or group access token...and possibly some other types as well.
         # See the GitLab Token Overview Documentation for info: https://docs.gitlab.com/ee/security/token_overview.html
-        gitlab_token = os.getenv("GITLAB_TOKEN")
-        if not gitlab_token:
+        gitlab_token = os.getenv("GITLAB_TOKEN", "")
+        if not gitlab_token and self.is_in_mr:
             raise SystemExit(" [!] A GitLab token with API access must be set at `GITLAB_TOKEN`")
         self._gitlab_token = gitlab_token
 
+    @property
+    def is_in_mr(self) -> bool:
```

Comment:
It is not necessary. There was a thought early on that the base class would have this property...but even then it wouldn't have to be a property. If it were a method, it could even be extracted from the class and possibly even decorated with an `lru_cache`. Might try this...

Reply:
When Python 3.8 is the minimum supported version, it will be possible to use the [`@cached_property` decorator](https://devdocs.io/python~3.10/library/functools#functools.cached_property) here...but it still doesn't need to be a property.